tests/Unit/Evolution/Systems/RelativisticEuler/Valencia/Test_FixConservatives.cpp

```diff
@@ -0,0 +1,116 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Determinant.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/RelativisticEuler/Valencia/FixConservatives.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "tests/Unit/TestCreation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+template <size_t Dim>
+void test_fix_conservatives(
+    const RelativisticEuler::Valencia::FixConservatives<Dim>& variable_fixer) {
+  Scalar<DataVector> tilde_d{DataVector{1.2, 9.e-13, 2.0, 0.0, 8.7}};
+  Scalar<DataVector> tilde_tau{DataVector{-0.001, 0.7, 0.1, 5.5, -0.05}};
+  auto tilde_s =
+      make_with_value<tnsr::i<DataVector, Dim, Frame::Inertial>>(tilde_d, 0.0);
+  for (size_t i = 0; i < Dim; ++i) {
+    tilde_s.get(i) = DataVector{1.25, -1.56, 4.32, 5.0, 0.0};
+  }
+  // Use identity matrix as metric for simplicity
+  auto inv_spatial_metric =
+      make_with_value<tnsr::II<DataVector, Dim, Frame::Inertial>>(tilde_d, 0.0);
+  for (size_t i = 0; i < Dim; ++i) {
+    inv_spatial_metric.get(i, i) = 1.0;
+  }
+  // assuming that determinant is correctly implemented,
+  // the determinant should equal its square root
+  auto sqrt_det_spatial_metric = determinant(inv_spatial_metric);
+
+  auto tilde_s_squared = make_with_value<Scalar<DataVector>>(tilde_d, 0.0);
+  dot_product(make_not_null(&tilde_s_squared), tilde_s, tilde_s,
+              inv_spatial_metric);
+
+  // Second and fourth elements must be set to D_min = 1.0e-12
+  Scalar<DataVector> expected_tilde_d_after_fixing{
+      DataVector{1.2, 1.0e-12, 2.0, 1.0e-12, 8.7}};
+
+  // First and fifth elements must be set to zero
+  Scalar<DataVector> expected_tilde_tau_after_fixing{
+      DataVector{0.0, 0.7, 0.1, 5.5, 0.0}};
+
+  auto expected_tilde_s_after_fixing = tilde_s;
+  const double safety_factor_for_s_tilde = 1.0e-11;
```

Comment:
Why are you using "s_tilde" here instead of "tilde_s" like everywhere else? This is surprising when reading the code.

Reply:
Good catch. Done.